lib/azuremetadata/azuremetadata.py

```diff
@@ -80,6 +80,12 @@ def list_api_versions(self):
         # considered all available APIs
         return self._get_api_newest_versions()
 
+    def set_api_version(self, api_version):
```

Comment:
This method should be used in `__init__` as well, that change should be a different PR IMO.